modules/ldapbackend/ldapauthenticator.cc

```diff
@@ -119,41 +142,45 @@ int LdapGssapiAuthenticator::attemptAuth( LDAP *conn )
   SaslDefaults defaults;
   char *ldapOption = 0;
 
-  ldap_get_option( conn, LDAP_OPT_X_SASL_MECH, ldapOption );
-  if ( !ldapOption )
+  int optret = ldap_get_option( conn, LDAP_OPT_X_SASL_MECH, &ldapOption );
+  if ( ( optret != LDAP_OPT_SUCCESS ) || !ldapOption )
     defaults.mech = std::string( "GSSAPI" );
   else
     defaults.mech = std::string( ldapOption );
   ldap_memfree( ldapOption );
+  ldapOption = 0;
```

Comment:
I'd prefer the use of `nullptr` over `0`, since it makes it obvious to the reader that we are dealing with a pointer.

Reply:
Sure, no problem